File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter08/ConsoleListener/SimpleCarLibrary/CarService.vb
'All the important enterprise service types are contained here.
Imports System.EnterpriseServices
Imports System.Runtime.InteropServices 'For GUID attribute
Imports System.Text 'For StringBuilder

<Guid("5A1BB09D-8D4B-481a-A441-DC6D8D48F396")> _
Public Interface IGetInfo
   Function GetInfo() as String
   Sub AnotherMethod()
End Interface

 'Class requires JIT activation
<JustInTimeActivation(), _
 EventTrackingEnabled(), _
 ClassInterface(ClassInterfaceType.None)> _
Public Class CarService
   Inherits ServicedComponent
   Implements IGetInfo

   Public Function GetInfo() As String Implements IGetInfo.GetInfo
      Dim ctxInfo As New StringBuilder()

      'Use ContextUtil to fetch context information
      ctxInfo.AppendFormat("Context ID:  {0}{1}", _
         ContextUtil.ContextId, vbCrLf)
      ctxInfo.AppendFormat("Activity ID: {0}{1}", _
         ContextUtil.ActivityId, vbCrLf)

      'If in transaction, get transaction ID
      Dim txId As String = "No Tx"
      If ContextUtil.IsInTransaction Then
         txId = ContextUtil.TransactionId.ToString()
      End If

      ctxInfo.AppendFormat("Transaction ID:    {0}{1}", txId, vbCrLf)
      ctxInfo.AppendFormat("Security Enabled?: {0}{1}", _
         ContextUtil.IsSecurityEnabled, vbCrLf)
         
      Return ctxInfo.ToString()
   End Function
   
   Public Sub AnotherMethod() Implements IGetInfo.AnotherMethod
   End Sub   
   
   Public Sub New
   End Sub   

End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter08/DisplayingMethods Right/SimpleCarLibrary/CarService.cs
using System;
using System.Diagnostics;              // event Logging
using System.EnterpriseServices;       // ServicedComponent
using System.Runtime.InteropServices;  // Class Interface
using System.Text;                     // StringBuilder
using System.Threading;

namespace SimpleCarLibrary
{
   [Guid("5A1BB09D-8D4B-481a-A441-DC6D8D48F396")]
   public interface IGetInfo
   {
      string GetInfo();
      void AnotherMethod();
   }

   [JustInTimeActivation(), 
    Guid("00E1160C-F65B-45fa-A92D-C24AB92831C8"),
    EventTrackingEnabled(),
    ClassInterface(ClassInterfaceType.None)]
   public class CarService : ServicedComponent, IGetInfo
   {
      public CarService()
      {
         string logMsg = string.Format("Created Object: {0}", this.GetHashCode());
         EventLog.WriteEntry("CarService.ctor()", logMsg);
      }

      public string GetInfo()
      {
         StringBuilder ctxInfo = new StringBuilder();
         
         // Use ContextUtil to fetch context information
         ctxInfo.AppendFormat("Context ID:  {0}\n", ContextUtil.ContextId);
         ctxInfo.AppendFormat("Activity ID: {0}\n", ContextUtil.ActivityId);
        
         // If in transaction, get transaction ID
         string txId = "No Tx";
         if (ContextUtil.IsInTransaction)
            txId = ContextUtil.TransactionId.ToString();

         ctxInfo.AppendFormat("Transaction ID:    {0}\n", txId);
         ctxInfo.AppendFormat("Security Enabled?: {0}\n", 
            ContextUtil.IsSecurityEnabled);

         return ctxInfo.ToString();
      }

      public void AnotherMethod() { }
   }
}